test/awaitable_senders_test.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <unifex/coroutine.hpp>
+
+#if !UNIFEX_NO_COROUTINES
+
+#  include <unifex/awaitable_sender.hpp>
+#  include <unifex/just.hpp>
+#  include <unifex/sync_wait.hpp>
+#  include <unifex/task.hpp>
+
+#  include <gtest/gtest.h>
+
+using namespace unifex;
+
+TEST(awaitable_senders, test) {
+  auto makeTask = [&]() -> task<std::optional<int>> {
+    co_return co_await just(42);
+  };
+
+  std::optional<std::optional<int>> answer =
+      sync_wait(awaitable_sender(makeTask()));
```

Comment:
As an aside, I get confused every time I see `awaitable_sender`.
Is it an awaitable type that wraps a sender or is it a sender that wraps an awaitable type?

Reply:
Me too. I suggest `as_awaitable` and `as_sender`.

Reply:
I'll rename things as a separate PR to keep this one narrowly focused.